app/models/measurement_details.rb

```diff
@@ -16,17 +16,22 @@ def initialize(attributes = {})
     if @id
       @measurement = Measurement.find_by(total_id: id) || Measurement.find_by_perm_link(id)
     end
+
+    # make sure instance vars are set
+    @current_power = Power.current
+    gr_singleton
+    ministry
   end
 
   def load
     validate!
 
-    load_total_from_gr
-    load_local_from_gr
-    load_user_from_gr
-    load_sub_mins_from_gr
-    load_team_from_gr
-    load_split_measurements
+    tasks = [:load_local_from_gr, :load_total_from_gr, :load_user_from_gr, :load_sub_mins_from_gr,
+             :load_team_from_gr, :load_split_measurements]
+    threads = tasks.map do |method|
+      Thread.new { send(method) }
+    end
+    threads.each(&:join)
```

Comment:
@draffensperger I figured out that a timeout here would not only by bad because stuck threads could get lost, but they are also unnecessary since the restclient should handle the timeout. Could I get a review on this PR?